quill-core/src/main/scala/io/getquill/sources/SourceMacro.scala

```diff
@@ -47,6 +47,21 @@ trait SourceMacro extends Quotation with ActionMacro with QueryMacro with Resolv
         runQuery(quotedTree, ast, inPlaceParams, params)(r, s, queryType(t.tpe))
     }
 
+  def runSingle[R, S](quoted: Expr[Quoted[Any]])(implicit r: WeakTypeTag[R], s: WeakTypeTag[S]): Tree = {
+    val t = c.WeakTypeTag(quoted.actualType.baseType(c.weakTypeOf[Quoted[Any]].typeSymbol).typeArgs.head)
```

Comment:
Is it possible to extract a method with the common parts between `run` and `runSingle`. I imagine the only difference would be a flag, right?